src/test/java/org/cactoos/iterator/ReversedTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterator;
+
+import org.cactoos.list.ListOf;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.hamcrest.core.IsEqual;
+import org.junit.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasSize;
+
+/**
+ * Test case for {@link Reversed}.
+ *
+ * @since 0.45
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+public final class ReversedTest {
+
+    @Test
+    public void reversesIterator() {
+        new Assertion<>(
+            "Must reverse the iterator",
+            new ListOf<>(
+                new Reversed<>(
+                    new IteratorOf<>("c", "a", "c", "t", "o", "o", "s")
+                )
+            ),
+            new IsEqual<>(
+                new ListOf<>(
```

Comment:
@marceloamadeu can't we directly compare iterators? or maybe use an existing matcher that allows to compare iterator?

Reply:
@victornoel I couldn't find another way to resolve it. Do you have any suggestion?
Thank!!!

Reply:
@marceloamadeu let's forget about it, thank you for trying :)